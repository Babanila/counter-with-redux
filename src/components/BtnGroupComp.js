import React from "react"
import { useDispatch } from "react-redux"
import { increment, decrement, cancel } from "../actions"
import "../styles/BtnGroupComp.css"

const ButtonGroup = () => {
  const dispatch = useDispatch()
  return (
    <div>
      <button onClick={() => dispatch(increment())}>+</button>
      <button onClick={() => dispatch(decrement())}>-</button>
      <button className="cancelButton" onClick={() => dispatch(cancel())}>
        reset
      </button>
    </div>
  )
}

export default ButtonGroup
