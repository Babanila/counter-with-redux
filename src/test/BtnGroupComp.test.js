import React from "react"
import { shallow } from "enzyme"
import BtnGroupComp from "../components/BtnGroupComp"

describe("BtnGroupComp Test", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<BtnGroupComp />)
  })

  it("should render and match the snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })

  it("should not log errors in console", () => {
    const spy = jest.spyOn(global.console, "error")
    expect(wrapper).not.toBeNull()
    expect(spy).not.toHaveBeenCalled()
  })

  it("should render the exact number of components", () => {
    expect(wrapper.find("div").length).toBe(1)
    expect(wrapper.find("button").length).toBe(3)
  })
})
