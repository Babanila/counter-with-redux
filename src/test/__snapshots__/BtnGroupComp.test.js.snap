// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BtnGroupComp Test should render and match the snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    className="cancelButton"
    onClick={[Function]}
  >
    reset
  </button>
</div>
`;
